runtime/lua/vim/diagnostic.lua

```diff
@@ -1796,6 +1800,26 @@ function M.open_float(opts, ...)
     end
   end
 
+  if message_kind == 'markdown' then
```

Comment:
Do I get this right that this takes the kind from the first diagnostic and then formats all messages accordingly?

What if some are text and others markdown?  That could cause to wrong highlighting

Reply:
Duh you're right. These diagnostics might not all have the same kind. I should instead set `message_kind` to `markdown` if _any_ one of them has Markdown content.

Reply:
> I should instead set message_kind to markdown if any one of them has Markdown content.

I think the highlighting should be restricted to the messages where message_kind is markdown. Highlighting plaintext as markdown can lead to incorrect styling. E.g. unquoted variables like `_foo` with a trailing `_` later in the text would highlight the whole section.

Reply:
Agree; that's the whole point of declaring the format: knowing when to parse/render and when not to.

Reply:
But then what if some diagnostics have `message_kind` set to `markdown` and others set to `plaintext`? They would all be displayed in the same floating window, so how do we highlight that?

I might be missing something here though.

Reply:
I'm not sure what's the best way to go about it, but we have to correlate diagnostics with the line numbers where the messages of them were rendered.

Or maybe the highlight application can be integrated directly into the loop here:

https://github.com/neovim/neovim/blob/4459e0cee8b6d043ab2b06cbd89545c45a76a612/runtime/lua/vim/diagnostic.lua#L1798-L1812

Otherwise it would need to track the number of lines used per diagnostic. For example if there are 3 diagnostics, where the second uses markdown you'd have a mapping of:

- 1-diagnostic: rendered in line 1-4,
- 2-diagnostic: rendered in line 5-6,
- 3-diagnostic: rendered in line 7

And then the ` api.nvim_buf_add_highlight` would only affect line 5-6

Reply:
https://github.com/neovim/neovim/pull/28412 might be useful for this(?)

Reply:
Oh yes, that does look handy 👀